source/SemVerAnalyzer/Nuget/NugetClient.cs

```diff
@@ -24,31 +25,26 @@ public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, string
 		{
 			try
 			{
-				using var client = new HttpClient();
-				
-				using var feedRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
-				using var feedResponse = await client.SendAsync(feedRequest);
-				if (!feedResponse.IsSuccessStatusCode)
-				{
-					var content = await feedResponse.Content.ReadAsStringAsync();
-					comments.Add($"Error retrieving Nuget feed:\n{content}");
+				var feedMeta = await GetUrlContents<NugetPackageFeed>(Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
+				if (feedMeta.Success == false){
+					comments.Add($"Error retrieving Nuget feed:\n{feedMeta.ErrorMessage}");
 					return null;
 				}
+					
+				var packageBaseAddress = feedMeta.Result.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
 
-				var feed = JsonSerializer.Deserialize<NugetPackageFeed>(await feedResponse.Content.ReadAsStringAsync());
-				var packageBaseAddress = feed.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
-				
-				using var versionRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(packageBaseAddress, packageName, "index.json"));
-				using var versionResponse = await client.SendAsync(versionRequest);
-				if (!versionResponse.IsSuccessStatusCode)
-				{
-					var content = await versionResponse.Content.ReadAsStringAsync();
-					comments.Add($"Error retrieving package versions:\n{content}");
+
+				var feedVersions = await GetUrlContents<VersionsFeed>(Path.Combine(packageBaseAddress, packageName, "index.json"));
+				if (feedMeta.Success == false){
+					comments.Add($"Error retrieving package versions:\n{feedMeta.ErrorMessage}");
 					return null;
 				}
 
-				var versions = JsonSerializer.Deserialize<VersionsFeed>(await versionResponse.Content.ReadAsStringAsync()).Versions;
-				var highestVersion = versions.OrderByDescending(v => v.MajorVersion()).ThenByDescending(v => v.MinorVersion()).ThenByDescending(v => v.PatchVersion()).First();
+				var highestVersion = feedVersions.Result.Versions
+					.Select(v => v.ToSemver())
+					.OrderBy(v => v)
+					.Last()
+					.ToString();
```

Comment:
FYI: OrderByDecending().First() is a couple of IL instructions shorter than OrderBy().Last() 😛